src/controllers/connectionManager.ts

```diff
@@ -496,6 +501,16 @@ export default class ConnectionManager {
 		}
 	}
 
+	public async handleSSLError(uri: string, profile: IConnectionProfile): Promise<IConnectionInfo> {
+		let updatedConn: IConnectionInfo | undefined;
+		this.showInstructionTextAsWarning(profile, async updatedConnection => {
```

Comment:
You aren't waiting for the callback to actually finish so this isn't going to work as expected (it will immediately move on and return updatedConn which is currently undefined). You need to await the call to showInstructionTextAsWarning

```suggestion
		await this.showInstructionTextAsWarning(profile, async updatedConnection => {
```

Reply:
Makes sense, previously it couldn't be awaited so I didn't update that after moving.. thanks for catching!